### PR TITLE
Phone Number Field Validation

### LINKED ISSUE #352: Storing phone number in model to follow internationally standardized format E.164
## Description
This is likely to be the most universal way to store phone numbers, given that the app users may have international numbers to work with. In the current model, `BigIntegerField()` has been used to store phone numbers. Adding proper validation to it or replacing it with proper phone number field should get the job done

## Mocks
contact

### Update [Required]
- [ ]Field shoud reject all inputs that doesn't follow format E.164 format

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor

### PR DESCRIPTION
**Pull request by Tahirih-Natang**
### Description
Added validation to phone number field to follow E.164 format. Also changed the form input type for contact number from 'number' to 'phone'

Fixes #352

### Type of Change:
- Code
- User Interface
  
**Code/Quality Assurance Only**
- Bug fix (non-breaking change which fixes an issue)

### CODE REVIEW

File: pcsa_GHN/models.py [hunk 1]
```diff
@@ -12,14 +10,22 @@
 
 class Contact(models.Model):
     office_name = models.CharField(max_length=200)
-    contact_number = models.BigIntegerField()
+    contact_number = models.CharField(
+        max_length=20,
```
Comment (sara-02) [inline pcsa_GHN/models.py]: @Tahirih-Natang correct me if I am wrong, but is not the `E.164` digit limit set to max 15, any reason why our max length is set to 20?

File: pcsa_GHN/models.py [hunk 2]
```diff
@@ -12,14 +10,22 @@
...
+        validators=[
+            RegexValidator(
+                r'^\s*(?:\+?(\d{1,3}))?([-. (]*(\d{3})[-. )]*)?((\d{3})[-. ]*(\d{2,4})(?:[-.x ]*(\d+))?)\s*$',
+                message="Please enter a valid phone number",
```
Comment (sara-02) [inline pcsa_GHN/models.py]: Would be great if we should also, so a sample phone number, along with the message, or something like `Please enter a valid phone number. Format <country code><number>`

File: pcsa_GHN/models.py [hunk 3]
```diff
@@ -12,14 +10,22 @@
 
 class Contact(models.Model):
     office_name = models.CharField(max_length=200)
-    contact_number = models.BigIntegerField()
+    contact_number = models.CharField(
+        max_length=20,
+        validators=[
+            RegexValidator(
```
Comment (sara-02) [inline pcsa_GHN/models.py]: @Tahirih-Natang why can't we use a simpler regex as shared in the Stackoverflow answer here https://stackoverflow.com/questions/6478875/regular-expression-matching-e-164-formatted-phone-numbers

Comment (Tahirih-Natang) [reply pcsa_GHN/models.py]: @sara-02 all changes have been made as requested

**Comment by sara-02** [comment]
@Tahirih-Natang thanks.
@medhach LGTM, please merge this PR :)

**Pull request merged**